lib/origen_debuggers/j_link.rb

```diff
@@ -341,11 +346,43 @@ def read_ir(reg_or_val, options = {})
 
     # Other methods can expose unique features of a given debugger
     module Custom
+      def set_interface(interface)
+        # set interface and reset
+        value = interface == :swd ? 1 : 0   # set interface : JTAG=0, SWD=1
+        dw "si #{value}"
+        # pull a reset now
+        dw 'RSetType 2' # reset via reset pin which should be same as manual reset pin
+        # toggle.  Also forces CPU to halt when it comes out of reset
+        dw 'r'          # reset and halts the device (prob not needed)
+        dw 'halt'       # halt core just in case
+      end
+
+      def halt
+        dw 'halt'
+      end
+
+      def quit
+        dw 'q'
+      end
+
       def read_memory(address, options = {})
         options = {
-          number_of_bytes: 1
+          number: 1,    # number of items to read
+          type:   :byte           # type of data to be read
         }.merge(options)
-        dw "mem 0x#{address.to_s(16).upcase}, #{options[:number_of_bytes]}"
+
+        # for backward compatibility
+        options[:number] = options[:number_of_bytes] if options[:number_of_bytes]
+
+        if options[:type] == :_32
```

Comment:
There are discrete memXX commands so can't really have user enter anything they want.   But could go numeric and just give error if invalid value.